inferno-vc/test/Spec.hs

```diff
@@ -0,0 +1,225 @@
+          o3 <- createObj $ MarkedBreakingWithPred h2
+          runOperation vcClientEnv (pushFunction o3) $ \h3 -> do
+            o4 <- createObj $ MarkedBreakingWithPred h3
+            runOperation vcClientEnv (pushFunction o4) $ \h4 -> do
+              -- Test fetchFunction:
+              forM_ [(o1, h1), (o2, h2), (o3, h3), (o4, h4)] $ \(o, h) ->
+                runOperation vcClientEnv (fetchFunction h) $ \o' -> do
+                  timestamp o' `shouldBe` timestamp o
+                  obj o' `shouldBe` obj o
+
+              -- Test fetchVCObject:
+              forM_ [(o1, h1), (o2, h2), (o3, h3), (o4, h4)] $ \(o, h) ->
+                runOperation vcClientEnv (fetchVCObject h) $ \o' ->
+                  case obj o' of
+                    VCFunction e t -> do
+                      timestamp o' `shouldBe` timestamp o
+                      (e, t) `shouldBe` (obj o)
+                    _ -> expectationFailure "Expected to get a VCFunction"
+
+              -- Test fetchVCObjects:
+              runOperation vcClientEnv (fetchVCObjects [h1, h3, h4]) $ \hashToMeta -> do
+                Set.fromList (Map.keys hashToMeta) `shouldBe` Set.fromList [h1, h3, h4]
+                forM_ [(o1, h1), (o3, h3), (o4, h4)] $ \(o, h) ->
+                  case Map.lookup h hashToMeta of
+                    Just meta ->
+                      timestamp meta `shouldBe` timestamp o
+                    Nothing -> expectationFailure "impossible"
+
+              -- fetchFunctionsForGroups only returns the head h4:
+              runOperation vcClientEnv (fetchFunctionsForGroups (Set.singleton 432)) $ \metas -> do
+                map obj metas `shouldBe` [h4]
+
+              -- The closure of h4 should be empty as it has no dependencies:
+              runOperation vcClientEnv (fetchVCObjectClosureHashes h4) $ \metas -> do
+                metas `shouldBe` []
+
+    it "deletion" $ do
+      o1 <- createObj Init
+      runOperation vcClientEnv (pushFunction o1) $ \h1 -> do
+        o2 <- createObj Init
+        runOperation vcClientEnv (pushFunction o2) $ \h2 -> do
+          o3 <- createObj Init
+          runOperation vcClientEnv (pushFunction o3) $ \h3 -> do
+            o4 <- createObj Init
+            runOperation vcClientEnv (pushFunction o4) $ \h4 -> do
+              runOperation vcClientEnv (deleteVCObject h3) $ \() -> do
+                -- Fetching h3 should fail:
+                runOperationFail vcClientEnv (fetchFunction h3) $ \case
+                  VCServerError (CouldNotFindPath _) -> pure ()
+                  _ -> expectationFailure ""
+                -- Others should fetch:
+                forM_ [(o1, h1), (o2, h2), (o4, h4)] $ \(o, h) ->
+                  runOperation vcClientEnv (fetchFunction h) $ \o' -> do
+                    timestamp o' `shouldBe` timestamp o
+                    obj o' `shouldBe` obj o
+
+    -- -- TODO is fetchFunctionsForGRoups wrong? It is returning deleted scripts:
```

Comment:
@siddharth-krishna is this something that needs to be addressed?

Reply:
It needs more investigation, but it is unrelated to this PR (if it is a bug it's an existing bug). Maybe I'll open an issue to track it?

Reply:
https://github.com/plow-technologies/inferno/issues/50